Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -403,11 +407,20 @@ open class _ComposerVC<ExtraData: ExtraDataTypes>: _ViewController,
     
     @objc open func shrinkInput(sender: UIButton) {
         Animate {
-            let leadingContainerSubviews = self.composerView.leadingContainer.subviews
-            for subview in leadingContainerSubviews where subview !== self.composerView.shrinkInputButton {
-                subview.isHidden = false
-            }
             self.composerView.shrinkInputButton.isHidden = true
+
+            let isNotShrinkInputButton = { (view: UIView) in
+                view !== self.composerView.shrinkInputButton
+            }
+            let showView = { (view: UIView) in view.isHidden = false }
+
+            self.composerView.leadingContainer.subviews
+                .filter(isNotShrinkInputButton)
+                .forEach(showView)
+
+            if self.channelController.channel?.config.uploadsEnabled == false {
+                self.composerView.attachmentButton.isHidden = true
+            }
```

Comment:
There's some dark magic happening here :) I must say I don't think that making the closures named helps the readability 🤔 The "normal" style is shorter and IMO clear enough:

```swift
self.composerView.leadingContainer.subviews
    .filter { $0 !== composerView.srinkInputButton }
    .forEach { $0.isHidden = false }
```

Reply:
Yes I totally exaggerated on this one 😄